opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequest.java

```diff
@@ -57,11 +58,25 @@ OpenSearchResponse search(Function<SearchRequest, SearchResponse> searchAction,
   @EqualsAndHashCode
   class IndexName {
     private static final String COMMA = ",";
+    private static final String COLON = ":";
 
+    private final String[] indexFullNames;
     private final String[] indexNames;
 
+    /**
+     * Constructor.
+     * indexNames are indexFullNames without the "{cluster}:" prefix.
+     */
     public IndexName(String indexName) {
-      this.indexNames = indexName.split(COMMA);
+      this.indexFullNames = indexName.split(COMMA);
+      // Remove all "<cluster>:" prefix if they exist
+      this.indexNames = Arrays.stream(indexFullNames)
+              .map(name -> name.substring(name.indexOf(COLON) + 1))
+              .toArray(String[]::new);
+    }
+
+    public String[] getIndexFullNames() {
```

Comment:
what is the definition of index full names?

Reply:
It's meant to distinguish indexNames and indexFullNames, where the indexFullNames includes the "<cluster>:" prefix, and indexNames remove such prefix and only have the actual name of each index.

I'm not very satisfied with the naming here. Any suggestions?

Reply:
I believe that index names are a high-level concept, and internally, OpenSearch uses cluster:index to specify an index. I have a question regarding SQL engines: Do they also utilize the cluster information?

Reply:
The SQL engine does not use the cluster name at all

Reply:
I believe that distinguishing the cluster name is unnecessary as it is an internal concept in OpenSearch that represents the index name.

Reply:
done